bezier-rs/lib/src/lib.rs

```diff
@@ -624,37 +640,43 @@ impl Bezier {
 
 	/// Implementation of the algorithm to find curve intersections by iterating on bounding boxes.
 	/// - `self_original_t_interval` - Used to identify the `t` values of the original parent of `self` that the current iteration is representing.
-	/// Note that the `t` interval the other curve is not needed since we want to return `t` with respect to it.
-	fn intersections_between_subcurves(&self, self_original_t_interval: [f64; 2], other: &Bezier, error: f64) -> Vec<f64> {
+	/// - `other_original_t_interval` - Used to identify the `t` values of the original parent of `other` that the current iteration is representing.
```

Comment:
Can you look into using [Range](https://doc.rust-lang.org/std/ops/struct.Range.html) instead of an array of length 2? I *think* this is the right use for that.